### PR TITLE
INTYGFV-15177: clientValidationError is not displayed as expected

### PR DESCRIPTION
**Pull request by SamuelTilly**
This aims to make `clientValidationErrors` easier to reason about.

Instead of creating `validationError` inside `DatePickerCustom` and having to create date handling in every Ue component this instead handle `clientValidationErrors` as part of the `updateCertificateDataElement` action in `certificateMiddleware.ts`. Depending on `config.type` and `value.type` we can create `clientValidationErrors`. A question is not updated until all client validation errors are resolved.

image

### CODE REVIEW

File: packages/webcert/src/feature/list/filter/DateRangeFilter.tsx [hunk 1]
```diff
@@ -166,18 +166,18 @@ const DateRangeFilter: React.FC<Props> = ({ config, onChange }) => {
...
+  //     if (validationError.field === configTo.id) {
+  //       setToValidationError(isInactive ? null : validationError)
+  //       dispatch(updateValidationError({ id: configTo.id, value: !isInactive }))
+  //     } else {
+  //       setFromValidationError(isInactive ? null : validationError)
+  //       dispatch(updateValidationError({ id: configFrom.id, value: !isInactive }))
+  //     }
+  //   },
+  //   [configFrom.id, configTo.id, dispatch]
+  // )
```
Comment (SopraErik) [inline packages/webcert/src/feature/list/filter/DateRangeFilter.tsx:180]: Maybe remove this commented out section?

**Review by SopraErik — CHANGES_REQUESTED** [review]
Really nice work!

Some bugs discovered when testing on certificates though

On checkboxDateList all date fields are marked as wrong format, when only one has an illegal date
image

Also, year component does not validate format error until attempting to sign, and this error does not disappear when the entry is corrected, not even when trying to sign

image

File: packages/webcert/src/feature/list/filter/DateRangeFilter.tsx (hunk 1 above)
Comment (SamuelTilly) [reply packages/webcert/src/feature/list/filter/DateRangeFilter.tsx]: Good catch, I had forgotten about the list validations, so this needed some more work. Should be fixed now.

**Comment by SamuelTilly** [comment]
> On checkboxDateList all date fields are marked as wrong format, when only one has an illegal date image
> 
> Also, year component does not validate format error until attempting to sign, and this error does not disappear when the entry is corrected, not even when trying to sign

This should hopefully be fixed now 😅

**Comment by SopraErik** [comment]
> > On checkboxDateList all date fields are marked as wrong format, when only one has an illegal date
> > Also, year component does not validate format error until attempting to sign, and this error does not disappear when the entry is corrected, not even when trying to sign
> 
> This should hopefully be fixed now 😅

In checkboxDateGroup the error message now disapears when checking another box in the list

image

File: packages/common/src/types/certificate.ts [hunk 2]
```diff
@@ -108,11 +139,11 @@ export interface CertificateDataConfig {
   header?: string
   icon?: string
   text: string
+  label?: string
   description: string
   type: ConfigTypes
   accordion?: ConfigAccordion
-
-  [propName: string]: unknown
+  list?: unknown
```
[the remark this answers is not in the document]
Comment (SamuelTilly) [reply packages/common/src/types/certificate.ts:146]: This was to combat a typing issue, when removing `[propName: string]: unknown` the type system becomes much pickier, for most part this was an easy fix but for list it became a bit trickier. Would like to remove list as well, but that can be addressed later.

**Comment by SamuelTilly** [comment]
> > > On checkboxDateList all date fields are marked as wrong format, when only one has an illegal date
> > > Also, year component does not validate format error until attempting to sign, and this error does not disappear when the entry is corrected, not even when trying to sign
> > 
> > 
> > This should hopefully be fixed now 😅
> 
> In checkboxDateGroup the error message now disapears when checking another box in the list
> 

Wops, I refactored `UeCheckboxDateGroup` and made it a bit simpler. Should work as expected now.

**Review by SopraErik — APPROVED** [review]
Looking really nice!

File: packages/common/src/types/certificate.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (ChrilleSigerhed) [reply packages/common/src/types/certificate.ts:146]: Okay, I understand.

**Pull request merged**